x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java

```diff
@@ -644,6 +645,111 @@ public void updateConnectorFilteringDraft(
+                ConnectorFiltering connectorFilteringSingleton = connectorFilteringList.get(0);
+
+                ConnectorFiltering activatedConnectorFiltering = connectorFilteringSingleton.setDraft(
+                    new FilteringRules.Builder().setRules(connectorFilteringSingleton.getDraft().getRules())
+                        .setAdvancedSnippet(connectorFilteringSingleton.getDraft().getAdvancedSnippet())
+                        .setFilteringValidationInfo(validation)
+                        .build()
+                );
+
+                final UpdateRequest updateRequest = new UpdateRequest(CONNECTOR_INDEX_NAME, connectorId).doc(
+                    new IndexRequest(CONNECTOR_INDEX_NAME).opType(DocWriteRequest.OpType.INDEX)
+                        .id(connectorId)
+                        .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+                        .source(Map.of(Connector.FILTERING_FIELD.getPreferredName(), List.of(activatedConnectorFiltering)))
+                );
+
+                client.update(updateRequest, new DelegatingIndexNotFoundActionListener<>(connectorId, l, (ll, updateResponse) -> {
+                    if (updateResponse.getResult() == UpdateResponse.Result.NOT_FOUND) {
+                        ll.onFailure(new ResourceNotFoundException(connectorNotFoundErrorMsg(connectorId)));
+                        return;
+                    }
+                    ll.onResponse(updateResponse);
+                }));
+            } catch (Exception e) {
+                l.onFailure(e);
+            }
+        }));
+
+    }
+
+    /**
+     * Activates the draft {@link ConnectorFiltering} property of a {@link Connector}.
+     *
+     * @param connectorId  Request for updating connector filtering property.
+     * @param listener     Listener to respond to a successful response or an error.
+     */
+    public void activateConnectorDraftFiltering(String connectorId, ActionListener<UpdateResponse> listener) {
+        getConnector(connectorId, listener.delegateFailure((l, connector) -> {
+            try {
+                List<ConnectorFiltering> connectorFilteringList = fromXContentBytesConnectorFiltering(
+                    connector.getSourceRef(),
+                    XContentType.JSON
+                );
+                // Connectors represent their filtering configuration as a singleton list
+                ConnectorFiltering connectorFilteringSingleton = connectorFilteringList.get(0);
+
+                FilteringValidationState currentValidationState = connectorFilteringSingleton.getDraft()
+                    .getFilteringValidationInfo()
+                    .getValidationState();
+
+                if (currentValidationState != FilteringValidationState.VALID) {
+                    throw new ElasticsearchStatusException(
+                        "Filtering draft needs to be validated by the framework before activation. "
```

Comment:
```suggestion
                        "Filtering draft needs to be validated by the connector service before activation. "
```

I think being explicit with "connector service" or "connector framework" is a bit better.